src/shared/bootspec.h

```diff
@@ -20,6 +20,21 @@ typedef enum BootEntryType {
         _BOOT_ENTRY_TYPE_INVALID = -EINVAL,
 } BootEntryType;
 
+typedef struct BootEntryAddon {
+        char *location;
+        char *cmdline;
+} BootEntryAddon;
+
+typedef struct BootEntryAddonArray {
+        BootEntryAddon *arr;
+        size_t size;
+} BootEntryAddonArray;
```

Comment:
This should just be called `BootEntryAddons`. The fact that it's an array is an implementation detail.

The first fields should be called something like `items`, and the counter should be called `count`, because it's not about the (allocated) size, so the old name is misleading.